### ISSUE TITLE
Removal of Faraday dependency breaks error handling

### ISSUE THREAD

**Issue by stevenringo** [intridea/omniauth-oauth2#73]
Commit `3be4ea`, release 1.3.0 refers.

Unless I am mistaken, the main OAuth2 gem as it stands (still) uses Faraday and removing Faraday breaks any error handling raised by Faraday in the OAuth2 gem.

**Comment by twelvelabs** [comment]
Yes, OAuth2 uses Faraday and it doesn't look like it's doing any error handling of it's own: https://github.com/intridea/oauth2/blob/0ab3c213f7baa43f232eb18fa46301e1a9460f7d/lib/oauth2/client.rb#L54

Are there plans to remove faraday from OAuth2, or is this a regression?
